### PR TITLE
partykit case weights

### PR DESCRIPTION
**Pull request by topepo**

### CODE REVIEW

File: R/partykit.R [hunk 1]
```diff
@@ -5,6 +5,10 @@
 #' arguments (as opposed to being specified in [partykit::ctree_control()]).
 #' @param formula A symbolic description of the model to be fit.
 #' @param data A data frame containing the variables in the model.
+#' @param weights A vector of weights whose length is the same as `nrow(data)`.
+#' For [partykit::ctree()] models, these are required to be non-negative
+#' integers while for [partykit::cforest()] they cna be non-negative integers
```
Comment (hfrick) [inline R/partykit.R:10]:
```suggestion
#' integers while for [partykit::cforest()] they can be non-negative integers
```

File: R/partykit.R [hunk 2]
```diff
@@ -70,14 +75,34 @@ ctree_train <-
         data = rlang::expr(data),
         !!!opts
       )
-    rlang::eval_tidy(tree_call)
+    if (!is.null(weights)) {
+      if (!is.vector(weights) ||
+          !is.integer(weights) ||
+          length(weights) != nrow(data)) {
+        rlang::abort("'weights' should be an integer vector with size the same as the rows of 'data'.")
```
Comment (hfrick) [inline R/partykit.R:82]:
```suggestion
        rlang::abort("'weights' should be an integer vector with size the same as the number of rows of 'data'.")
```

File: R/partykit.R [hunk 3]
```diff
@@ -123,8 +148,31 @@ cforest_train <-
...
         data = rlang::expr(data),
+        weights = rlang::expr(weights),
         !!!opts
       )
+
+    if (!is.null(weights)) {
+      if (!is.vector(weights) ||
+          !is.numeric(weights) ||
+          length(weights) != nrow(data)) {
+        rlang::abort("'weights' should be a numeric vector with size the same as the rows of 'data'.")
```
Comment (hfrick) [inline R/partykit.R:159]:
```suggestion
        rlang::abort("'weights' should be a numeric vector with size the same as the number of rows of 'data'.")
```

File: tests/testthat/test_partykit.R [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                         mincriterion = 1/2, maxdepth = 2,
+                         weights = 1:nrow(Chicago)),
+    regex = NA
+  )
+  expect_false(isTRUE(all.equal(fit_2$fitted, fit_3$fitted)))
+  expect_error(
+    fit_4 <- ctree_train(Class ~ ., data = ad_data),
+    regex = NA
+  )
+  expect_error(
```
Comment (hfrick) [inline tests/testthat/test_partykit.R:33]: this could be a snapshot test now that parsnip uses testthat 3e

File: tests/testthat/test_partykit.R [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    fit_4 <- cforest_train(Class ~ ., data = ad_data, ntree = 2),
+    regex = NA
+  )
+  expect_error(
+    fit_5 <- cforest_train(Class ~ ., data = ad_data, ntree = 2, mtry = 2000),
+    regex = NA
+  )
+  expect_equal(fit_5$info$control$mtry, 130)
+  expect_error(
```
Comment (hfrick) [inline tests/testthat/test_partykit.R:78]: this also could be a snapshot test now

**Review by hfrick — APPROVED** [review]

**Review by EmilHvitfeldt — APPROVED** [review]

**Pull request merged**